src/rtctools_heat_network/heat_physics_mixin.py

```diff
@@ -1462,6 +1538,274 @@ def __pipe_heat_to_discharge_path_constraints(self, ensemble_member):
+            stored_heat = self.state(f"{ates}.Stored_heat")
+            heat_loss = self.state(f"{ates}.Heat_loss")
+
+            flow_dir_var = self._pipe_to_flow_direct_map[hot_pipe]
+            is_buffer_charging = self.state(flow_dir_var)
+
+            sup_carrier = parameters[f"{ates}.T_supply_id"]
+            ret_carrier = parameters[f"{ates}.T_return_id"]
+            supply_temperatures = self.temperature_regimes(sup_carrier)
+            return_temperatures = self.temperature_regimes(ret_carrier)
+            ates_temperature = self.state(f"{ates}.Temperature_ates")
+            ates_temperature_disc = self.state(f"{ates}__temperature_ates_disc")
+            ates_dt_charging = self.state(f"{ates}.Temperature_change_charging")
+            ates_dt_loss = self.state(f"{ates}.Temperature_loss")
+            ates_temperature_loss_nominal = self.variable_nominal(f"{ates}.Temperature_loss")
+
+
+            if len(supply_temperatures) == 0:
+                constraints.append((ates_dt_charging, 0.0, 0.0))
+                constraints.append((ates_dt_loss, 0.0, 0.0))
+                coeff_efficiency_ates = parameters[f"{ates}.heat_loss_coeff"]
+                constraints.append(((heat_loss - stored_heat*coeff_efficiency_ates)/heat_loss_nominal, 0.0, 0.0))
+            else:
+                big_m_dt = bounds[f"{ates}.Temperature_loss"][1] #max(supply_temperatures)
+                big_m_heatloss = 2*heat_loss_nominal
+                for ates_temperature in supply_temperatures:
+                    ates_temperature_is_selected = self.state(f"{ates}__temperature_disc_{ates_temperature}")
+                    #setting temperature losses to zero when lowest discrete temperature is selected, does not work because then the ates will just not be used.
+                    # if ates_temperature == min(supply_temperatures):
+                    #     constraints.append(((ates_dt_loss + big_m_dt*(1-ates_temperature_is_selected))/ates_temperature_loss_nominal, 0.0, 0.0))
+                    # else:
+                    #     #if is selected, then specific temeprature loss constraint should be applicable, which will be a function of the stored heat
+                    #     # a(T_disc-Tamb) exp(-storedheat*b) where a and b are factors and the exponential function is linearised.
+                    a,b = self.__get_linear_Tloss_vs_storedheat(heat_stored_max, ates_temperature, temperature_ambient=soil_temperature)
+                    stored_heat_vec = ca.repmat(stored_heat, len(a))
+                    ates_dt_loss_vec = ca.repmat(ates_dt_loss, len(a))
+                    ates_temperature_is_selected_vec = ca.repmat(ates_temperature_is_selected, len(a))
+                    #TODO: improve nominal
+                    constraints.append(
+                        (( ates_dt_loss_vec - (a*stored_heat_vec/heat_stored_max + b) + big_m_dt*(1-ates_temperature_is_selected_vec))/ates_temperature_loss_nominal, 0.0 , np.inf)
```

Comment:
I think this constraint might be wrong:
- Due to the inequality we might end up with positive losses in temperature, which are physically not possible

I think this cannot be done with inequality and should be a single linear equality constraint or piece-wise linear